src/main/java/org/akaza/openclinica/control/core/SecureController.java

```diff
@@ -670,6 +665,15 @@ else if (currentRole.getId() > 0
             mayProceed();
             // pingJobServer(request);
             // Set if enrollment is capped. Used by navBar.jsp to hide "Add Participant" link in the menu
+
+            //If user logged in using an expired state cookie, bring them to the login redirect.
+            if (RequestContextHolder.getRequestAttributes().getAttribute("StateCookieInvalid", RequestAttributes.SCOPE_SESSION) != null) {
```

Comment:
create a static final variable for storing the name of the attribute so it can be referenced in all places including the class where you are setting this attribute. You may want to create that variable in the request authenticator class as a public variable and reference it here.